### ISSUE TITLE
Simulation model contains partial models

### ISSUE THREAD

**Issue by casella** [lbl-srg/modelica-buildings#2230]
The compilation of [Buildings.Fluid.HeatPumps.Calibration.ScrollWaterToWater](https://libraries.openmodelica.org/branches/newInst/Buildings_7.0.0/files/Buildings_7.0.0_Buildings.Fluid.HeatPumps.Calibration.ScrollWaterToWater.err) fails in OpenModelica with this error
```
[Buildings 7.0.0/Fluid/HeatPumps/Compressors/BaseClasses/PartialCompressor.mo:2:1-170:22:writable] Notification: From here:
[Buildings 7.0.0/Fluid/HeatPumps/BaseClasses/PartialWaterToWater.mo:132:3-136:26:writable] Error:
Component heaPum.com has partial type PartialCompressor.
```
In fact, when looking at the redeclaration of [heaPum](url)
https://github.com/lbl-srg/modelica-buildings/blob/0dead434179c75f011b45ff804986e842d9bc712/Buildings/Fluid/HeatPumps/Calibration/ScrollWaterToWater.mo#L8
the component `com` is not redeclared. Being partial, I guess it should. 

@perost, can you confirm that?

**Comment by perost** [comment]
No, the ScrollWaterToWater itself contains a redeclare of `com`, this is a bug in our check for partial elements. I've fixed it in [#6906](https://github.com/OpenModelica/OpenModelica/pull/6906).

**Comment by casella** [comment]
Aha, I missed that, [here it is](https://github.com/lbl-srg/modelica-buildings/blob/0dead434179c75f011b45ff804986e842d9bc712/Buildings/Fluid/HeatPumps/ScrollWaterToWater.mo#L7).

Thanks @perost for the quick feedback
